### PR TITLE
log: Dump signature request properly.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Currently percent encoded strings are not properly encoded.

`s3%!!(MISSING)A(MISSING)`

Print properly as json encoded.

`"query":{"events":["s3:ObjectCreated:*","s3:ObjectRemoved:*","s3:ObjectAccessed:*"]`

## Motivation and Context
Friendly message.

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/utils.go [hunk 1]
```diff
@@ -246,12 +245,12 @@ func dumpRequest(r *http.Request) string {
...
-		Query  string      `json:"query"`
+		Query  url.Values  `json:"query"`
 		Header http.Header `json:"header"`
-	}{r.Method, r.URL.Path, r.URL.RawQuery, header}
-	jsonBytes, err := json.Marshal(req)
+	}{r.Method, r.URL.Path, r.URL.Query(), header}
+	jsonBytes, err := json.Marshal(&req)
 	if err != nil {
-		return "<error dumping request>"
+		return string(fmt.Sprintf("%#v", req))
```
Comment (krisis) [inline cmd/utils.go]: `fmt.Sprintf` returns `string`. `string()` is superfluous.

**Review by krisis — CHANGES_REQUESTED** [review]

File: cmd/utils.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/utils.go]: oops yes!

Comment (harshavardhana) [reply cmd/utils.go]: Fixed.

File: cmd/utils_test.go [hunk 2]
```diff
@@ -413,3 +414,42 @@ func TestCheckURL(t *testing.T) {
 		}
 	}
 }
+
+// Testsing dumping request function.
```
Comment (krisis) [inline cmd/utils_test.go]: typo: s/Testsing/Testing/

**Review by krisis — COMMENTED** [review]
LGTM, minor typo.

**Review by krishnasrinivas — APPROVED** [review]

**Pull request merged**